specifyweb/frontend/js_src/lib/components/WorkBench/CellMeta.ts

```diff
@@ -46,6 +46,386 @@ const defaultMetaValues = Object.freeze([
+        0
+      ),
+      invalidCells: cellMeta.reduce(
+        (count, info) =>
+          count +
+          (this.getCellMetaFromArray(info, 'issues').length > 0 ? 1 : 0),
+        0
+      ),
+      searchResults: cellMeta.reduce(
+        (count, info) =>
+          count + (this.getCellMetaFromArray(info, 'isSearchResult') ? 1 : 0),
+        0
+      ),
+      modifiedCells: cellMeta.reduce(
+        (count, info) =>
+          count + (this.getCellMetaFromArray(info, 'isModified') ? 1 : 0),
+        0
+      ),
+    };
+    
+    // below line breaks error cell highlighting
+    // this.workbench.setCellCounts(this.cellCounts);
+
+    // OLD CODE: NEEDS TO BE REMOVED AT THE END
+    // Update navigation information
+    // needs to be changed
+    // Object.values(
+    //   this.wbView.el.getElementsByClassName('wb-navigation-total')
+    // ).forEach((navigationTotalElement) => {
+    //   const navigationContainer = navigationTotalElement.closest(
+    //     '.wb-navigation-section'
+    //   );
+    //   if (navigationContainer === null) return;
+    //   const navigationType = navigationContainer.getAttribute(
+    //     'data-navigation-type'
+    //   ) as keyof WbCellCounts | null;
+    //   if (navigationType === null) return;
+    //   navigationTotalElement.textContent =
+    //     cellCounts[navigationType]?.toString();
+
+    //   if (cellCounts[navigationType] === 0) {
+    //     const currentPositionElement =
+    //       navigationContainer.getElementsByClassName(
+    //         'wb-navigation-position'
+    //       )?.[0];
+    //     if (typeof currentPositionElement === 'object')
+    //       currentPositionElement.textContent = '0';
+    //   }
+    // });
+
+    // const uploadButton =
+    //   this.wbView.el.querySelector<HTMLButtonElement>('.wb-upload');
+    // if (uploadButton === null) return;
+    // const title = wbText.uploadUnavailableWhileHasErrors();
+    // if (
+    //   !uploadButton.disabled ||
+    //   uploadButton.getAttribute('title') === title
+    // ) {
+    //   const hasErrors = cellCounts.invalidCells > 0;
+    //   uploadButton.toggleAttribute('disabled', hasErrors);
+    //   uploadButton.setAttribute('title', hasErrors ? title : '');
+    // }
+  }
+
+public cellIsType(metaArray: WbMetaArray, type: keyof WbCellCounts): boolean {
+    switch (type) {
+      case 'invalidCells': {
+        return this.getCellMetaFromArray(metaArray, 'issues').length > 0;
+      }
+      case 'newCells': {
+        return this.getCellMetaFromArray(metaArray, 'isNew');
+      }
+      case 'modifiedCells': {
+        return this.getCellMetaFromArray(metaArray, 'isModified');
+      }
+      case 'searchResults': {
+        return this.getCellMetaFromArray(metaArray, 'isSearchResult');
+      }
+      default: {
+        return false;
+      }
+    }
+  }
+}
+/* eslint-enable functional/no-this-expression */
+
+
 /* eslint-disable functional/no-this-expression */
 export class WbCellMeta {
```

Comment:
I left the old code behind so it's easier for me to look at the old version and see what I'm missing. Will remove it and rename everything at the end. Sorry, didn't realize it would make reviewing harder!

Reply:
Removed all old code except WbView - will do that after I go over it once to make sure I didn't miss anything.